gtsfm/averaging/translation/averaging_1dsfm.py

```diff
@@ -108,4 +123,97 @@ def run(
         for i in range(num_images):
             if wRi_list[i] is not None and wti_values.exists(i):
                 wti_list[i] = wti_values.atPoint3(i)
-        return wti_list
+
+        # Compute the metrics.
+        if gt_wTi_list is not None:
+            ta_metrics = _compute_metrics(inliers, outliers, i2Ui1_dict, wRi_list, wti_list, gt_wTi_list)
+        else:
+            ta_metrics = None
+
+        return wti_list, ta_metrics
+
+
+def _get_measurement_angle_errors(
+    i1_i2_pairs: Tuple[int, int],
+    i2Ui1_measurements: Dict[Tuple[int, int], Unit3],
+    gt_i2Ui1_measurements: Dict[Tuple[int, int], Unit3],
+) -> List[float]:
+    """Returns a list of the angle between i2Ui1_measurements and gt_i2Ui1_measurements for every 
+    (i1, i2) in i1_i2_pairs.
+
+    Args:
+        i1_i2_pairs: List of (i1, i2) tuples for which the angles must be computed.
+        i2Ui1_measurements: Measured translation direction of i1 WRT i2.
+        gt_i2Ui1_measurements: Ground truth translation direction of i1 WRT i2.
+
+    Returns:
+        List of angles between the measured and ground truth translation directions.
+    """
+    errors = []
+    for (i1, i2) in i1_i2_pairs:
+        if (i1, i2) in i2Ui1_measurements and (i1, i2) in gt_i2Ui1_measurements:
+            errors.append(
+                comp_utils.compute_relative_unit_translation_angle(
+                    i2Ui1_measurements[(i1, i2)], gt_i2Ui1_measurements[(i1, i2)]
+                )
+            )
+    return errors
+
+
+def _compute_metrics(
```

Comment:
i think i'm fine with having this function in this file, but I'm curious what our long term strategy is for where to put these metric related functions (that are specific to a module). Or maybe this is the only one.